### ISSUE TITLE
Cleaning the notebook erases undo history

### ISSUE THREAD

**Issue by Taytay** [fastai/nbdev-vscode#18]
This makes sense, but code formatters don't seem to have this problem. Could this be done in a similar way somehow, so that I can keep the ability to undo?

**Comment by hamelsmu** [comment]
I'm not sure tbh are you able to undo the similar action in a jupyter notebook after a clean operation?

**Comment by rebornix** [comment]
Once we finalize `onWillSaveNotebookDocument` in VS Code, maybe we could listen to this event and return edits for it so VS Code can handle the edit and also undo/redo for it automatically. 

It seems right now nbdev is editing the ipynb file directly, is this correct @hamelsmu ?

**Comment by hamelsmu** [comment]
That's correct I didn't realize there was a way to submit edits I'll look into that

**Comment by Taytay** [comment]
> I'm not sure tbh are you able to undo the similar action in a jupyter notebook after a clean operation?

I'm too new to the world of Notebooks to know! This is my first time cleaning them :)
